src/header/DataSelector.js

```diff
@@ -12,57 +13,56 @@ type Props = $ReadOnly<{
 }>;
 
 export default function DataSelector(props: Props): React$Node {
-	function onChange(ev: SyntheticEvent<HTMLInputElement>) {
-		const fileInput = ev.currentTarget;
-		if (fileInput.files && fileInput.files[0]) {
-			const reader = new FileReader();
-			reader.onload = (buffer: ProgressEvent) => {
-				const reader = buffer.currentTarget;
-
-				if (
-					!(reader instanceof FileReader) ||
-					typeof reader.result !== 'string'
-				) {
-					throw new Error();
-				}
-
-				let result;
-				try {
-					result = JSON.parse(reader.result);
-				} catch (ex) {
-					console.error(ex);
-					alert('The custom level_data JSON is invalid.');
-					return;
-				}
+	async function openFile() {
+		const blob = await fileOpen();
+		const reader = new FileReader();
+		reader.onload = (buffer: ProgressEvent) => {
+			const reader = buffer.currentTarget;
+			if (
+				!(reader instanceof FileReader) ||
+				typeof reader.result !== 'string'
+			) {
+				throw new Error();
+			}
 
-				props.onNewLevelsLoad(result);
-			};
-			reader.onerror = (ex) => {
+			let result;
+			try {
+				result = JSON.parse(reader.result);
+			} catch (ex) {
 				console.error(ex);
-				alert('There was a problem reading the file.');
-			};
-			reader.readAsText(fileInput.files[0]);
-		}
+				alert('The custom level_data JSON is invalid.');
+				return;
+			}
+
+			props.onNewLevelsLoad(result);
+		};
+		reader.onerror = (ex) => {
+			console.error(ex);
+			alert('There was a problem reading the file.');
+		};
+		reader.readAsText(blob);
 	}
 
-	function saveFile() {
-		downloadBlob(
-			new Blob([JSON.stringify(props.levels)], {
-				// must be this type to ensure no file extension
-				type: 'application/octet-stream',
-			}),
-			'level_data'
-		);
+	async function saveFile() {
+		const blob = new Blob([JSON.stringify(props.levels)]);
+		try {
+			await fileSave(blob, {
+				fileName: 'level_data',
+			});
+		} catch (err) {
+			if (err.name !== 'AbortError') {
+				return console.error(err);
+			}
+			console.log('The user aborted a request.');
+		}
 	}
 
 	return (
 		<>
 			<div className={styles.space}>
-				<CustomFileInput onChange={onChange}>
-					<button type="button" tabIndex={-1}>
-						Load custom level_data
-					</button>
-				</CustomFileInput>
+				<button type="button" tabIndex={-1} onClick={openFile}>
```

Comment:
`tabindex={-1}` was previously done due to how the custom file input is implemented, the real file input is overlayed on top and made invisible, while this button is fake, we don't want keyboard navigation to actually tab into this fake button.

Since we are now using the new module, this button is real now
```suggestion
				<button type="button" onClick={openFile}>
```